### ISSUE TITLE
Password protection not working?

### ISSUE THREAD

**Issue by magdkudama** [srikanth-lingala/zip4j#20]
Hello,

I'm trying to create a password protected ZIP file, but I keep getting that the password is not correct when I try to unzip the contents:

```java
var params1 = new ZipParameters();
params1.setEncryptionMethod(EncryptionMethod.ZIP_STANDARD);
params1.setCompressionLevel(CompressionLevel.NORMAL);
params1.setCompressionMethod(CompressionMethod.DEFLATE);
params1.setEncryptFiles(true);
params1.setFileNameInZip("file1.csv");
params1.setEncryptFiles(true);

var params2 = new ZipParameters();
params2.setEncryptionMethod(EncryptionMethod.ZIP_STANDARD);
params2.setCompressionLevel(CompressionLevel.NORMAL);
params2.setCompressionMethod(CompressionMethod.DEFLATE);
params2.setEncryptFiles(true);
params2.setFileNameInZip("file2.csv");
params2.setEncryptFiles(true);

var zip = new ZipFile("MY_ZIP.zip", "hello".toCharArray());
zip.addStream(s1, params1);
zip.addStream(s2, params2);

zip.getFile();
// Check the file...
```

This issue seems related: https://github.com/srikanth-lingala/zip4j/issues/6, but I'm using version 2.1.0, so it should be fixed now?

I'll report if I find anything else.

Thank you.

**Comment by magdkudama** [comment]
I can confirm that without a password, it's working just fine. The problem is when setting a password

**Comment by srikanth-lingala** [comment]
out of curiosity: Does it work fine with AES encryption?

**Comment by magdkudama** [comment]
I added to both `params1` and `params2`:

```java
setEncryptionMethod(EncryptionMethod.AES);
```

Then I tried extracting with `7z`, and only the first file could be extracted? The rest were marked as: `ERROR: Wrong password`

Could it be due to the fact I'm adding multiple streams with multiple `ZipParameters`? I can try with one file and STANDARD encryption.

**Comment by magdkudama** [comment]
Ups, I'm so sorry, I'm stupid. I forgot to remove the `EncryptionMethod.STANDARD` from all files.

To reply to your question: *YES, it does work with AES encryption*, so it's something to do with standard encryption that's failing.

**Comment by srikanth-lingala** [comment]
I just tried to reproduce this issue. And yes, I agree with your findings: `ZipFile.addStreams()` works with AES encryption and not with ZipStandard.

I have also identified the problem, and will include a fix soon.

**Comment by magdkudama** [comment]
Thank you @srikanth-lingala

**Comment by srikanth-lingala** [comment]
Fixed in v2.1.1 which is released today
